### PR TITLE
Get hyperlinks for the texts/paragraphs

### PR DESCRIPTION
**Pull request by rabi92**
### This PR includes
- Get hyperlinks from the text/paragraphs
- Get the hyperlink type if its an asset type or an external type.
- Optional property of a hyperlink in the content type to exist only if there is an hyperlink. _paragraph.content.hyperlink_. 
Please refer to the updated schema [here](https://github.com/rabi92/airppt-models/blob/dev/pptelement.ts#L75).

Each paragraph has an array of contents. this content object within the paragraph may have an hyperlink property

Reference to OXML Documentation: http://officeopenxml.com/drwSp-text-runProps.php

### CODE REVIEW

File: ts/parsers/paragraphparser.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { getValueAtPath } from "../helpers";
```
Comment (shak) [inline ts/parsers/paragraphparser.ts]: just noticed it now but do these files need to be suffixed with 'parser'? 

`ts/parsers/paragraph.ts` will suffice no? 

At the very least, I'd expect camelcasing, `ts/parsers/paragraphParser.ts`

File: ts/parsers/paragraphparser.ts [hunk 2]
```diff
@@ -13,25 +13,62 @@ import {
...
+                if (
+                    contents[i].text[0].trimEnd().length === contents[i].text[0].length &&
+                    contents[i + 1].text[0].trimStart().length === contents[i + 1].text[0].length
+                ) {
+                    contents[i].text[0] += " " + contents[i + 1].text[0];
+                } else {
+                    contents[i].text[0] += contents[i + 1].text[0];
+                }
+                contents.splice(i + 1, 1);
+                i--;
```
Comment (shak) [inline ts/parsers/paragraphparser.ts:36]: this took me a while to get my head around .. 😅

File: ts/parsers/relparser.ts [hunk 3]
```diff
@@ -15,32 +15,36 @@ export default class SlideRelationsParser {
...
-        return linkDetails;
+        return this.getRelationDetails(relID);
+    }
+
+    public static resolveParagraphHyperlink(element): Content["hyperlink"] {
+        const relID = getValueAtPath(element, '["a:rPr"][0]["a:hlinkClick"][0]["$"]["r:id"]');
+        if (!relID) {
+            return null;
+        }
+        return this.getRelationDetails(relID);
```
Comment (shak) [inline ts/parsers/relparser.ts:31]:
```suggestion

        return this.getRelationDetails(relID);
```

File: ts/parsers/relparser.ts [hunk 4]
```diff
@@ -15,32 +15,36 @@ export default class SlideRelationsParser {
...
             if (relationDetails["Id"] == relID) {
-                let linkType = LinkType.Asset;
+                let linkType: LinkType;
                 if (relationDetails["TargetMode"] && relationDetails["TargetMode"] === "External") {
                     linkType = LinkType.External;
                 } else {
                     linkType = LinkType.Asset;
                 }
-                let relElement: PowerpointElement["links"] = {
+                return {
```
Comment (shak) [inline ts/parsers/relparser.ts:46]:
```suggestion

                return {
```

File: ts/parsers/paragraphparser.ts [hunk 5]
```diff
@@ -13,25 +13,62 @@ import {
  * Parse the paragraph elements
  */
 export default class ParagraphParser {
+    //Merge consecutive text content blocks together which have same hyperlinks
```
Comment (shak) [inline ts/parsers/paragraphparser.ts:18]:
```suggestion
    //Merge consecutive text content blocks together which have same hyperlinks
    //and also adjust the spacing in between the hyperlink for the edge cases
```

**Review by shak — APPROVED** [review]
LGTM, minor things

File: ts/parsers/paragraphparser.ts (hunk 1 above)
Comment (rabi92) [reply ts/parsers/paragraphparser.ts]: They were named originally like that, didn't change, however the scripts i created i did made them camel case, going to change them.

**Pull request merged**